deviseの実装

zorori777 opened the pull request:
# WHAT
deviseのview,controllerを導入

# WHY
ユーザーーの情報やviewを表示するため

zorori777 committed:
[WIP]deviseの実装

zorori777 committed:
devise周りの導入

# WHAT
・deviseファイルを適応
・edit.erbをhamlに変換
・userのsassファイルをimport
# WHY
・表示させるため
・sassを当てるためapplication.scssにimport

zorori777 committed:
usesコントローラの作成

# WHAT
コントローラに作成

# WHY
・ルーティングの対してactionを定義するため
・対応するviewを表示させるため

takebnn26 commented on app/views/devise/registrations/edit.html.haml:
```diff
@@ -0,0 +1,39 @@
+\#account-page.account-page
```
ん？これってコメントアウトされてる？

takebnn26 replied:
あとここじゃなくてusers_controllerのedit使ってほしいなあ

takebnn26 replied:
あとこの画面につながるようなボタンを他の場所に作ってほしいなあ

takebnn26 commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,2 @@
+class UsersController < ApplicationController
```
これはdeviseのレジストレーションのedit使うんじゃなくて、users_controllerのedit使うようにしよう！

takebnn26 commented on app/views/devise/registrations/new.html.haml:
```diff
@@ -0,0 +1,32 @@
+#account-page.account-page
```
あとこの画面につながるようなボタンを他の場所に作ってほしいなあ

takebnn26 commented on app/views/devise/sessions/new.html.haml:
```diff
@@ -0,0 +1,21 @@
+#account-page.account-page
```
あとこの画面につながるようなボタンを他の場所に作ってほしいなあ

zorori777 committed:
[WIP]  deviseのセット

# WHAT
　ボタンログイン、ログアウト、新規登録機能の追加
# WHY
 userの管理をするため

zorori777 commented:
＜devise関連＞
authenticate_user! :ユーザーがログインしているかを確認してくれる。before_actionと組み合わせて、ログインしていないときに強制的にindexに働かせる機能などができる。
user_sighed_in?:現在サインインしているユーザーを取得。主に条件分岐の際に利用する。
current_user:現在ログインしているユーザーを取得できる。

takebnn26 commented on app/controllers/users_controller.rb:
```diff
@@ -1,2 +1,6 @@
 class UsersController < ApplicationController
+
+  def edit
```
@user定義してん

takebnn26 commented on app/views/users/edit.html.haml:
```diff
@@ -0,0 +1,38 @@
+      = link_to 'ログアウト', destroy_user_session_path, method: :delete, rel: 'nofollow', class: 'btn'
+      = link_to 'トップページに戻る', root_path, class: 'btn'
+    .account-page__inner--right.user-form
+      = form_for (:user) do |f|
```
@userで使おう

takebnn26 commented on app/controllers/users_controller.rb:
```diff
@@ -1,2 +1,6 @@
+
+  def edit
+  end
+
```
updateアクションないけどアップデートされてる？

takebnn26 commented on app/views/messages/_flash-messages.html.haml:
```diff
@@ -0,0 +1,4 @@
+.flash-messsages
```
これはなんだ

takebnn26 replied:
ここもっと変えよう

zorori777 committed:
user update

# WHAT
・userの編集機能の追加
・validation :name_badge: ,email,password追加
# WHY
userを情報を編集し、保存するため

takebnn26 commented on app/controllers/users_controller.rb:
```diff
@@ -1,6 +1,24 @@
+    end
+
+    def user_setting
+      @user =User.find(params[:id])
```
=の後マス開けよう

takebnn26 commented on app/models/user.rb:
```diff
@@ -1,4 +1,11 @@
 class User < ApplicationRecord
+
+validates :email, presence: true
```
emailとpasswordのバリデーションはdeviseでやってるので書く必要ないね

takebnn26 commented on app/controllers/users_controller.rb:
```diff
@@ -1,6 +1,24 @@
+      if current_user.update_with_password(update_params)
+         redirect_to root_path,notice: '編集成功しました'
+      else
+        redirect_to edit_user_path,alert: '編集失敗しました'
```
renderにしよう
render とredirect_toの違いは調べよう

zorori777 replied:
<　render　>
contorollerで処理した結果の出力先をviewを指定する。
controllerのインスタンス変数は、そのままViewに渡される。

< redirect_to>
指定したcontroller、actionに再度リクエストを送信される。
controllerのインスタンス変数は、リダイレクト先で新たに生成される。

zorori777 committed:
user update(2)

# WHAT
validationのemail、passwordの記述削除
update内の記述編集
flash-messageのファイル作成
# WHY
validationはdevise内で記述されているため
renderとredirect_toの違い、flash-messageの記述変更

takebnn26 commented on app/controllers/users_controller.rb:
```diff
@@ -8,7 +8,8 @@ def update
          redirect_to root_path,notice: '編集成功しました'
       else
-        redirect_to edit_user_path,alert: '編集失敗しました'
+        flash.now[:alert] = "編集を失敗しました"
```
いいね！すばらしいい！

takebnn26 commented on app/controllers/users_controller.rb:
```diff
@@ -19,6 +20,6 @@ def update_params
 
     def user_setting
-      @user =User.find(params[:id])
+      @user = User.find(params[:id])
```
ナイス！これからも見やすいコードを意識していこう！

takebnn26 commented on app/controllers/users_controller.rb:
```diff
@@ -8,7 +8,8 @@ def update
       if current_user.update_with_password(update_params)
```
これ@user使っていいよ
せっかく定義してるから

takebnn26 commented on app/views/messages/_flash-messages.html.haml:
```diff
@@ -1,4 +1,4 @@
-  %p.flash-message__alert
+.flash-message
+  -flash.each do |key, value|
+    %div{ class: "alert alert-#{key}" }
```
もっと短く書けるね調べて実装しよう

zorori777 committed:
user update3

# WHAT
・ user_controller内のcurrent_user.updateから@user.updateに変更。
・content_tagにて記述

zorori777 committed:
flash-message css

# WHAT
flash-messageのcss

zorori777 commented:
![](https://i.gyazo.com/db26d825c9d509e2e8628f831bbc837d.png)

![](https://i.gyazo.com/28e6a54151e7288dd7a0fe2f56cd202c.png)

![](https://i.gyazo.com/d963ec5333035011cf9a04d766066689.png)



![](https://i.gyazo.com/2f3d19470d8711e117d48653a74573b9.png)

zorori777 committed:
追加
